### PR TITLE
Splints

### PR DESCRIPTION
**Pull request by jroinc**
## About The Pull Request

Splints are now *mostly* time-based. Previously, a splint applied by a medic had 45 health, while a splint applied by a PFC/SL had... zero health.

Now, applying a splint gives 5 minutes of time per level of medical, with a base time of 5 minutes (PFC gets 5, SL gets 10, Medic gets 15, and so on). This is reduced by 5 minutes if self-applied, to a minimum of 5 minutes.

If it's been three or more seconds since you've last moved, the timer's paused. If you're concerned about your splint breaking, don't move around unless you need to. Due to the way the tickrate works, the minimum timer decrement is 4 seconds, even if you just move one step.

Additionally, getting hit on a splint now removes 10 seconds from the splint per hit, so a full HP PFC splint can take 30 and a medic splint can take 90, rather than the previous... 1 and 4-5 respectively. This seems like a lot, but it's constantly decreasing as the timer's ticking down.

When the timer gets within 1 minute of breaking, every time it decrements has a 20% chance to warn you of this.

## Why It's Good For The Game

Splints are less RNG insta-break, but also not something you can just slap on and fight with forever. IMO, it's an overall buff.

## Changelog
:cl:
balance: Splints are now time-based. PFC splints last 5 min, medic splints last 15 min. The timer only counts when moving.
/:cl:

### CODE REVIEW

File: code/modules/organs/limbs.dm [hunk 1]
```diff
@@ -19,7 +19,7 @@
 	var/last_dam = -1
 	var/supported = FALSE
 	///How many instances of damage the limb can take before its splints fall off
```
Comment (hyper2snyper) [inline code/modules/organs/limbs.dm:21]: probably wanna change the autodoc.

File: code/modules/organs/limbs.dm [hunk 2]
```diff
@@ -558,6 +558,17 @@ Note that amputating the affected organ does in fact remove the infection from t
 	if((limb_status & LIMB_ROBOT)) //Robotic limbs don't heal or get worse.
 		return
 
+	if(limb_status & LIMB_SPLINTED) //Splints gradually break over time.
+		if(splint_time <= 0)
+			remove_limb_flags(LIMB_SPLINTED)
+			to_chat(owner, span_userdanger("The splint on your [display_name] comes apart!"))
+		else
+			if(world.time - last_move_time > 30)
```
Comment (hyper2snyper) [inline code/modules/organs/limbs.dm:566]: you didnt test this did you?

**Review by hyper2snyper — COMMENTED** [review]
Looks pretty good. Is this a buff or nerf?

File: code/modules/organs/limbs.dm (hunk 2 above)
Comment (jroinc) [reply code/modules/organs/limbs.dm:566]: I literally just made this. It still needs testing. I'm like 80% sure it doesn't compile.

File: code/modules/organs/limbs.dm (hunk 1 above)
Comment (jroinc) [reply code/modules/organs/limbs.dm:21]: damm, ok

File: code/modules/organs/limbs.dm (hunk 2 above)
Comment (jroinc) [reply code/modules/organs/limbs.dm]: fixed

File: code/modules/organs/limbs.dm (hunk 1 above)
Comment (jroinc) [reply code/modules/organs/limbs.dm]: fixed

**Comment by jroinc** [comment]
> 
> 
> Looks pretty good. Is this a buff or nerf?

IMO, it's a buff. Ungas go from being able to take 1 hit (2-3 if a medic applies it) on a splint, to 30-90 if in active combat... but that counter drops slowly over time.

**Comment by jroinc** [comment]
Tested, functions as intended. Needs some large-scale testing for general reactions before merg is considered.

**Comment by DeltaFire15** [comment]
xenos (except warrior punch) needing **90** hits to a **single limb** to break fresh medic splints seems very much overkill. While buffing splint survivability in general a bit seems nice (single throw of some ammo box breaking is it kinda yikes), this imo goes way too far in the other direction, making broken bones effectively meaningless (which marine doesn't carry splints and also does not have someone that carries some nearby)

**Comment by jroinc** [comment]
> 
> 
> xenos (except warrior punch) needing **90** hits to a **single limb** to break fresh medic splints seems very much overkill. While buffing splint survivability in general a bit seems nice (single throw of some ammo box breaking is it kinda yikes), this imo goes way too far in the other direction, making broken bones effectively meaningless (which marine doesn't carry splints and also does not have someone that carries some nearby)

One, Warrior punch still hard-kills a splint in one hit. Two, that counter's always dropping.

I considered just *removing* damage doing anything to splints, but I left a 10-second timer reduction as a compromise.

**Comment by DeltaFire15** [comment]
> One, Warrior punch still hard-kills a splint in one hit. Two, that counter's always dropping.
> 
> I considered just _removing_ damage doing anything to splints, but I left a 10-second timer reduction as a compromise.

Yes, I did state warrior's punch as example for the only thing viable against splints after this change for a reason.
The specific issue is "why is the in-field temporary solution infinitely more resistant to combat than the actual limb itself pre-injurity"
With this, it might actually be preferrable to *never* get bone breaks treated since now you effectively have a hyper-effective wound armor on that limb.

The counter automatically slowly decreasing is effectively almost-irrelevant considering one medic splint alone holds for fifteen minutes on its own.

**Review by BraveMole — CHANGES_REQUESTED** [review]
What delta fire said seems logic, is this not encouraging people to pre splint? 
Aren't you better off with a splint than without?

**Comment by jroinc** [comment]
> What delta fire said seems logic, is this not encouraging people to pre splint? Aren't you better off with a splint than without?

Do they not cause slowdown anymore?

File: code/modules/organs/limbs.dm [hunk 3]
```diff
@@ -558,6 +558,17 @@ Note that amputating the affected organ does in fact remove the infection from t
...
 		return
 
+	if(limb_status & LIMB_SPLINTED) //Splints gradually break over time.
+		if(splint_time <= 0)
+			remove_limb_flags(LIMB_SPLINTED)
+			to_chat(owner, span_userdanger("The splint on your [display_name] comes apart!"))
+		else
+			if(world.time - owner.last_move_time > 30)
+				return
+			splint_time = max(splint_time - 1, 0) //splint health decrements by one per lifetick, unless it's been more than 10 seconds since your last move.
```
Comment (Walarks) [inline code/modules/organs/limbs.dm:568]: Could just reduce like `splint_time -= 1` and then add `splint_time = 0` where you remove the flag.

**Comment by BraveMole** [comment]
closing as stale

**Pull request closed without merging**